clang/lib/Sema/SemaStmt.cpp

```diff
@@ -3967,6 +3968,16 @@ bool Sema::DeduceFunctionTypeFromReturnExpr(FunctionDecl *FD,
     // Update all declarations of the function to have the deduced return type.
     Context.adjustDeducedFunctionResultType(FD, Deduced);
 
+  // Emit a remark for deduced return type if the option fdump-auto-type-inference is enabled
+  SourceManager &SM = getSourceManager();
+  SourceLocation Loc = FD->getLocation();
+  if (SM.isWrittenInMainFile(Loc) &&
+      opts::DumpAutoTypeInference.getNumOccurrences()) {
+    DiagnosticsEngine &Diag = Context.getDiagnostics();
+    unsigned DiagID = Diag.getCustomDiagID(DiagnosticsEngine::Remark, "return type of function '%0' deduced as %1");
+    Diag.Report(Loc, DiagID) << FD->getNameAsString() << Deduced;
+  }
+
```

Comment:
Why is this code duplicated? Can you move this into a single source and call that function appropriately?